lib/hllcounter.cc

```
@@ -424,6 +424,8 @@ void HLLCounter::consume_fasta(
                 // Iterate through the reads and consume their k-mers.
                 try {
                     read = parser->get_next_read();
+                } catch (khmer::StreamReadError) {
+                    break;
```

I would let this be thrown and wrap it into a Python exception in `_khmer.cc`